core/src/test/scala/org/apache/spark/scheduler/PoolSuite.scala

```diff
@@ -178,4 +180,97 @@ class PoolSuite extends SparkFunSuite with LocalSparkContext {
     scheduleTaskAndVerifyId(2, rootPool, 6)
     scheduleTaskAndVerifyId(3, rootPool, 2)
   }
+
+  test("SPARK-18066: FIFO Scheduler just uses root pool") {
+    sc = new SparkContext("local", "PoolSuite")
+    val taskScheduler = new TaskSchedulerImpl(sc)
+
+    val rootPool = new Pool("", SchedulingMode.FIFO, initMinShare = 0, initWeight = 0)
+    val schedulableBuilder = new FIFOSchedulableBuilder(rootPool)
+
+    val taskSetManager0 = createTaskSetManager(stageId = 0, numTasks = 1, taskScheduler)
+    val taskSetManager1 = createTaskSetManager(stageId = 1, numTasks = 1, taskScheduler)
+
+    val properties = new Properties()
+    properties.setProperty("spark.scheduler.pool", TEST_POOL)
+
+    // FIFOSchedulableBuilder just uses rootPool so even if properties are set, related pool
+    // (testPool) is not created and TaskSetManagers are added to rootPool
```

Comment:
Can you re-phrase this as what *should* happen, which makes these comments easier to read? e.g. 

"When task sets are submitted, they should be added to the root pool, and no additional pools should be created (even though there's a configured default pool)."